core/src/main/kotlin/xtdb/trie/HashTrie.kt

```diff
@@ -23,42 +23,61 @@ interface HashTrie<N : Node<N>, L : N> {
 
     val leaves get() = rootNode?.leaves ?: emptyList()
 
+    val LEVEL_BITS: Int
```

Comment:
given that these now seem to be 'travelling together' everywhere, it's probably time to make them a struct. can then give the struct the 'bucketFor' methods for more idiomatic OO.

(comment maybe a bit vague, lmk and it's probably a quick pairing session)